orte/mca/rml/ofi/rml_ofi_send.c

```diff
@@ -0,0 +1,651 @@
+   
+ 	snd = OBJ_NEW(orte_rml_send_t);
+    snd->dst = *peer;
+    snd->origin = *ORTE_PROC_MY_NAME;
+    snd->tag = tag;
+	if (NULL != req->send.iov) {
+        snd->iov = req->send.iov;
+        snd->count = req->send.count;
+        snd->cbfunc.iov = req->send.cbfunc.iov;
+    } else {
+        snd->buffer = req->send.buffer;
+        snd->cbfunc.buffer = req->send.cbfunc.buffer;
+    }
+    snd->cbdata = req->send.cbdata;
+
+    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s send_msg_transport to peer %s at tag %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         ORTE_NAME_PRINT(peer), tag);
+
+	
+	/* get the peer address by doing modex_receive 	*/
+    opal_output_verbose(10, orte_rml_base_framework.framework_output,
+                         "%s calling OPAL_MODEX_RECV_STRING ", ORTE_NAME_PRINT(ORTE_PROC_MY_NAME) );
+    switch ( orte_rml_ofi.ofi_conduits[conduit_id].fabric_info->addr_format)
+    {
+       case  FI_SOCKADDR_IN :
+             OPAL_MODEX_RECV_STRING(ret, OPAL_RML_OFI_FI_SOCKADDR_IN, peer , (char **) &dest_ep_name, &dest_ep_namelen);
+             break;
+       case  FI_ADDR_PSMX :
+             OPAL_MODEX_RECV_STRING(ret, OPAL_RML_OFI_FI_ADDR_PSMX, peer , (char **) &dest_ep_name, &dest_ep_namelen);
+             break;
+    }
+    opal_output_verbose(10, orte_rml_base_framework.framework_output,
+                         "%s  Return value from OPAL_MODEX_RECV_STRING - %d, length returned - %d", 
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME), ret, dest_ep_namelen);
+    /*print the sockaddr - port and s_addr */
+	struct sockaddr_in* ep_sockaddr = (struct sockaddr_in*) dest_ep_name;
+	opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                       "%s obtained for peer %s port = 0x%x, InternetAddr = 0x%x  ",
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ORTE_NAME_PRINT(peer),ep_sockaddr->sin_port,
+                        ep_sockaddr->sin_addr.s_addr);
+
+	if ( OPAL_SUCCESS == ret) {
+		opal_output_verbose(10, orte_rml_base_framework.framework_output,
+                         "%s OPAL_MODEX_RECV succeded, %s peer ep name obtained. length=%d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         ORTE_NAME_PRINT(peer), dest_ep_namelen);
+		ret = fi_av_insert(orte_rml_ofi.ofi_conduits[conduit_id].av, dest_ep_name,1,&dest_fi_addr,0,NULL);
+		if( ret != 1) {	
+			opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s fi_av_insert failed in send_msg() returned %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),ret );
+			/* call the send-callback fn with error and return, also return failure status */
+			snd->status = ORTE_ERR_ADDRESSEE_UNKNOWN;
+			ORTE_RML_SEND_COMPLETE(snd);
+			snd = NULL;
+			//OBJ_RELEASE( ofi_send_req);
```

Comment:
If you fail to insert the address into the AV, is that a recoverable error?

I'm guessing that this should be an `opal_show_help()` / exit kind of error...?